rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -527,7 +535,17 @@ def decode_transaction_hashes(
                 )
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
-        for tx_hash in tx_hashes:
+        for tx_index, tx_hash in enumerate(tx_hashes):
```

Comment:
move chain_id_name out of the loop
```suggestion
        chain_name = self.evm_inquirer.chain_id.to_name()
        for tx_index, tx_hash in enumerate(tx_hashes):
```

So that it does not run the `to_name() all the time in the loop + 1 in the end.

... or perhaps even better. Change `evm_inquirer.chain_name` to be derived from chain_id and not blockchain and use that?